frontend/explorer/public/js/hash.js

```diff
@@ -1209,25 +1209,100 @@ function populateHashPage(hash, explorerHash) {
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendExplorerBlock(infoBody, explorerHash.block);
 	} else if (hashType === "transactionid") {
+		appendNavigationMenuTransaction(explorerHash.transaction);
 		appendHeading(infoBody, 'Hash Type: Transaction ID');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendTransactionStatistics(infoBody, explorerHash.transaction, explorerHash.unconfirmed!==true);
 		appendRawTransaction(infoBody, explorerHash.transaction.rawtransaction);
 	} else if (hashType === "unlockhash") {
+		appendNavigationMenuUnlockHash(hash);
 		appendHeading(infoBody, 'Hash Type: Unlock Hash');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendUnlockHashTables(infoBody, hash, explorerHash);
 	} else if (hashType === "coinoutputid") {
+		appendNavigationMenuCoinOutput(explorerHash, hash);
 		appendHeading(infoBody, 'Hash Type: Coin Output ID');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendCoinOutputTables(infoBody, hash, explorerHash);
 	} else if (hashType === "blockstakeoutputid") {
+		appendNavigationMenuBlockstakeOutput(explorerHash.transactions, hash);
 		appendHeading(infoBody, 'Hash Type: BlockStake Output ID');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendBlockStakeOutputTables(infoBody, hash, explorerHash);
 	}
 }
 
+// appendNavigationMenuTranscaction adds the transaction link to the top navigation menu
+function appendNavigationMenuTransaction(explorerTransaction) {
+	appendNavigationMenuBlock(explorerTransaction);
+	var navigation = document.getElementById('nav-links');
+	var transactionSpan = document.createElement('span');
+	navigation.appendChild(document.createTextNode(' > '));
+	navigation.appendChild(transactionSpan);
+	linkHash(transactionSpan, explorerTransaction.id, 'Transaction');
+}
+
+// appendNavigationMenuCoinOuput adds the coin output link to the top navigation menu
+function appendNavigationMenuCoinOutput(explorerHash, hash) {
+	//Coin Ouput
+	if (explorerHash.transactions != null) {
+		for (var i = 0; i < explorerHash.transactions.length; i++) {
+			for (var j = 0; j < explorerHash.transactions[i].coinoutputids.length; j++) {
+				if (explorerHash.transactions[i].coinoutputids[j] == hash) {
+					appendNavigationMenuTransaction(explorerHash.transactions[i]);
+					var navigation = document.getElementById('nav-links');
+					var outputSpan = document.createElement('span');
+					navigation.appendChild(document.createTextNode(' > '));
+					navigation.appendChild(outputSpan);
+					linkHash(outputSpan, explorerHash.transactions[i].coinoutputids[j], 'Coin Output');
+				}
+			} 
+		}
+	//Coin Ouput - Block Creator Reward
+	} else if (explorerHash.blocks != null) {
+		for (var i = 0; i < explorerHash.blocks.length; i++) {
```

Comment:
move this out of the else if, and instead just always do it. As long as the logic above returns when it is finished, this logic will never be reached.

Reply:
```js
if (explorerHash.transactions != null) {
    // Coin Ouput
    for (var i = 0; i < explorerHash.transactions.length; i++) {
        for (var j = 0; j < explorerHash.transactions[i].coinoutputids.length; j++) {
            if (explorerHash.transactions[i].coinoutputids[j] == hash) {
                appendNavigationMenuTransaction(explorerHash.transactions[i]);
                var navigation = document.getElementById('nav-links');
                var outputSpan = document.createElement('span');
                navigation.appendChild(document.createTextNode(' > '));
                navigation.appendChild(outputSpan);
                linkHash(outputSpan, explorerHash.transactions[i].coinoutputids[j], 'Coin Output');
                return;
            }
        } 
    }
}

if (explorerHash.blocks == null) {
    return;
}
//Coin Ouput - Block Creator Reward
for (var i = 0; i < explorerHash.blocks.length; i++) {
    for (var j = 0; j < explorerHash.blocks[i].minerpayoutids.length; j++) {
        if (explorerHash.blocks[i].minerpayoutids[j] == hash) {
            appendNavigationMenuBlock(explorerHash.blocks[i]);
            var navigation = document.getElementById('nav-links');
            var outputSpan = document.createElement('span');
            navigation.appendChild(document.createTextNode(' > '));
            navigation.appendChild(outputSpan);
            linkHash(outputSpan, explorerHash.blocks[i].minerpayoutids[j], 'Payout ID');
            return;
        }
    }
}
```

Reply:
Added return statement
Removed else if